src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -766,19 +766,69 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
     }
 
     void Schedule::handleGRUPNET(HandlerContext& handlerContext) {
+        auto network = this->snapshots.back().network.get();
+        std::vector<Network::Node> nodes;
         for (const auto& record : handlerContext.keyword) {
-            const auto& groupName = record.getItem("NAME").getTrimmedString(0);
-
-            if (!this->snapshots.back().groups.has(groupName))
-                addGroup(groupName , handlerContext.currentStep);
-
-            int table = record.getItem("VFP_TABLE").get< int >(0);
+            const std::string& groupNamePattern = record.getItem<ParserKeywords::GRUPNET::NAME>().getTrimmedString(0);
+            const auto group_names = this->groupNames(groupNamePattern);
+            if (group_names.empty())
+                this->invalidNamePattern(groupNamePattern, handlerContext);
+            const auto& pressure_item = record.getItem<ParserKeywords::GRUPNET::TERMINAL_PRESSURE>();
+            const int vfp_table = record.getItem<ParserKeywords::GRUPNET::VFP_TABLE>().get<int>(0);
 
-            auto new_group = this->snapshots.back().groups.get( groupName );
-            if (new_group.updateNetVFPTable(table))
-                this->snapshots.back().groups.update( std::move(new_group) );
+            for (const auto& group_name : group_names) {
+                const auto& group = this->snapshots.back().groups.get(group_name);
+                const auto& parent_name = group.parent();
+                if (!parent_name.empty())
+                {
+                    const std::string& downtree_node = group_name;
+                    const std::string& uptree_node = parent_name;
+                    if (vfp_table == 0) {
+                        // If vfp table is defaulted (or set to 0) then the group is not part of the network
+                        if (network.has_node(downtree_node) && network.has_node(uptree_node)) 
```

Comment:
I think `drop_branch` will check the `if` conditions, it is okay just to `drop_branch` without the if condition?

Reply:
It seems that `drop_branch` does not check these conditions. Omitting the `if` conditions would lead to a `No such node` throw.

Reply:
Okay. I think it is safer to check for now. I think the function `drop_branch` also needs some improvements.  there is some logic unclear in the Network side regarding the relationship of `node` and `network`. For the situation you mentioned, the `drop_branch` should eventually not doing anything, because there is no branch to drop yet. 

But I do not think it should be handled in this PR.